### PR TITLE
[stable10] implement MetaFileVersionNode::getMountPoint

### LINKED ISSUE #31797: Version thumbnails don't appear any more
### Steps

1. Login as "admin"
1. Create a file "test.txt" and edit several times to create versions
1. Open the "Versions" tab in the sidebar
1. Look at preview thumbnails

### Expected results
Thumbnails match

### Actual result
Thumbnail shows broken image, Internal server error.

### Version
stable10, happens after https://github.com/owncloud/core/pull/31785

### Log
```
{"reqId":"Uub58ileGe1ug8H1VAoT","level":4,"time":"2018-06-15T10:22:40+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"webdav","method":"GET","url":"\/owncloud\/remote.php\/dav\/meta\/15\/v\/1529057520?preview","message":"Exception: : {\"Exception\":\"OCP\\\\Files\\\\NotPermittedException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"
#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/PreviewManager.php(207): OC\\\\Files\\\\Node\\\\AbstractNode->getMountPoint()\\n
#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Files\\\/PreviewPlugin.php(111): OC\\\\PreviewManager->isAvailable(Object(OC\\\\Files\\\\Meta\\\\MetaFileVersionNode))\\n
#2 [internal function]: OCA\\\\DAV\\\\Files\\\\PreviewPlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n
#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n
#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(281): Sabre\\\\DAV\\\\Server->exec()\\n
#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n
#8 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/remote.php(165): require_once('\\\/srv\\\/www\\\/htdocs...')\\n
#9 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Node\\\/AbstractNode.php\",\"Line\":84}"}
```

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Missing implementation caused preview generation to fail

## Related Issue
Fixes #31797

## How Has This Been Tested?
- manual
- acceptance tests shall be added ?

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
acceptance tests documented as sub-item of https://github.com/owncloud/QA/issues/488

@DeepDiver1975 unit tests or not possible due to staticness?

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by DeepDiver1975** [comment]
WHOT?

```
PHP Fatal error: Call to a member function getMountPoint() on boolean in /drone/src/lib/private/Files/Storage/Common.php on line 712

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php [hunk 1]
```diff
@@ -463,7 +469,13 @@ public function getVersion($internalPath, $versionId) {
 			if ($stat === false) {
 				throw new NotFoundException();
 			}
-			return $this->objectStore->getVersion($this->getURN($stat['fileid']), $versionId);
+			$version = $this->objectStore->getVersion($this->getURN($stat['fileid']), $versionId);
+			list($uid, $path) = $this->convertInternalPathToGlobalPath($internalPath);
+			if (!empty($version)) {
```
Comment (PVince81) [inline lib/private/Files/ObjectStore/ObjectStoreStorage.php]: assuming version is never the string "0"...

**Comment by PVince81** [comment]
Looks good now.

Does this fix other issues we haven't discovered yet ? It seems you have added things that looked like were missing ? (things to add to the versions test plan perhaps https://github.com/owncloud/QA/issues/488)

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php (hunk 1 above)
Comment (DeepDiver1975) [reply lib/private/Files/ObjectStore/ObjectStoreStorage.php]: $version is an array

Comment (PVince81) [reply lib/private/Files/ObjectStore/ObjectStoreStorage.php]: oh... why is the method not called `getVersions()` in plural ? this is what confused me.
maybe it's an associative array, ok

**Pull request merged**